docs/developer/basic-form-support.md

```diff
@@ -0,0 +1,289 @@
+# Basic Form Support
+
+**NOTE:** This feature is under heavy development. Some of the described below may change in the future.
+
+Since Kubeapps 1.6.0, it's possible to attach a chart with a JSON schema that defines the structure of the `values.yaml` file. This JSON schema is used with two goals:
+
+ - Validate that the given values satisfy the schema defined. In case the submitted values are not valid, the installation or upgrade will fail.
+ - Present the user with a simple (basic) form so the chart is easier to deploy and configure.
+
+The goal of the basic form is to define the most common parameters that users typically modify before deploying a chart (like username and password) so they are able to modify them with a more user-friendly form.
+
+This document specifies what's needed to be defined in order to present this basic form to the users of a chart.
+
+## Create a values.schema.json
+
+This file, introduced with Helm v3, is a [JSON Schema](https://json-schema.org/) that defines the structure of the `values.yaml` file of the chart, including as many validations as needed. If a chart includes its schema, the values used are validated before submitting the new release.
+
+This file can define some or every possible value of the chart. Once it's written it should be included in the Helm package. The proposal to include it in Helm can be found [here](https://github.com/helm/helm/issues/5812).
+
+## Additional annotations used to identify basic parameters
+
+In order to identify, among all the values specified in the schema, the values that should be presented in the form, it's necessary to include some special tags.
+
+First of all, it's necessary to specify the tag `form`. This field should be unique and it should identify the parameter. All the properties marked with this key in the schema will be represented in the form. For example:
+
+```
+    "wordpressUsername": {
+      "type": "string",
+      "form": "username"
+    },
+```
+
+With the definition above, we are marking the value `wordpressUsername` as a value to be represented in the form (with the ID `username`). Note that we are also specifying the `type` that will be used to render the proper HTML components to represent the input in the form:
```

Comment:
Isn't `type` part of the jsonschema, and it's also being used here to render? In which case "Note that the `type` is also used to render..." (the current sentence implies that we're only specifying it for the rendering, rather than because it's part of the schema).

Reply:
okay, yes, it's used for both